src/mapblock.cpp

```diff
@@ -108,6 +109,72 @@ MapBlock::~MapBlock()
 		delete[] data;
 }
 
+NodeWithDef MapBlock::getNodeWithDefNoEx(v3s16 p)
+{
+	if(!isValidPosition(p))
+		return NodeWithDef(MapNode(CONTENT_IGNORE), m_gamedef->ndef());
+	return getNodeWithDefNoCheck(p, NULL);
+}
+
+HybridPtr<const ContentFeatures> MapBlock::getNodeDefNoCheck(v3s16 p, bool *valid_position)
+{
+	*valid_position = isValidPosition(p);
+	if(!valid_position) {
+		return NULL;
+	}
+	// If special definition exists, return a shared pointer to a copy of it
+	if(m_special_nodedefs.count(p))
+		return new ContentFeatures(m_special_nodedefs[p]);
```

Comment:
@Thomas--S May the pointer outlive the `MapBlock`? If not, why not return `m_special_nodedefs[p]` itself?